hypertrace-bom/build.gradle.kts

```diff
@@ -0,0 +1,31 @@
+plugins {
+  `java-platform`
+  id("org.hypertrace.publish-plugin")
+}
+
+javaPlatform {
+  allowDependencies()
+}
+
+dependencies {
+  api(platform(libs.grpc.bom))
+  api(platform(libs.jackson.bom))
+  api(platform(libs.hypertrace.kafka.bom))
```

Comment:
I am trying to understand the difference between catalog and platform. With catalog, each project will explicitly select their dependencies. Is this possible with this implementation? From the structure, this looks as though all the libraries in this project will get added as dependencies, but a particular service for e.g. may not be using lombok or guava. Is my understanding correct?

Reply:
> From the structure, this looks as though all the libraries in this project will get added as dependencies, but a particular service for e.g. may not be using lombok or guava. Is my understanding correct?

No. These are just adding constraints. i.e. if anyone depends on say grpc-core:1.46.0, but have this platform bom decalared, then the version will be resolved to the latest of 1.46.0 and what this bom defines(in this case 1.57.2).
Constraints are not dependencies added by default.

Reply:
https://docs.gradle.org/current/userguide/dependency_constraints.html#sec:adding-constraints-transitive-deps